ext/ffi/jit_trampoline.rs

```diff
@@ -58,11 +58,24 @@ fn native_to_c(ty: &NativeType) -> &'static str {
 
 pub(crate) fn codegen(sym: &crate::Symbol) -> String {
   let mut c = String::from(include_str!("prelude.h"));
-  let ret = native_to_c(&sym.result_type);
+  let needs_unwrap = matches!(
+    &sym.result_type,
+    NativeType::Function
+      | NativeType::Pointer
+      | NativeType::I64
+      | NativeType::ISize
+      | NativeType::U64
+      | NativeType::USize
+  );
+
+  // Return type of the FFI call.
+  let ffi_ret = native_to_c(&sym.result_type);
+  // Return type of the trampoline.
+  let ret = if needs_unwrap { "void" } else { ffi_ret };
 
   // extern <return_type> func(
   c += "\nextern ";
```

Comment:
Nitpick suggestion:
```rust
let _ = write!(c, "\nextern {ffi_ret} func(");
```
May be better to rename `ffi_ret` -> `return_type`, and possibly to just remove the comment at that point?